### PR TITLE
Rejigger the the cell backup so it blocks the ioloop less

### PR DESCRIPTION
**Pull request by jnwatson**

### CODE REVIEW

**Review by invisig0th — APPROVED** [review]

File: synapse/lib/cell.py [hunk 1]
```diff
@@ -1202,75 +1201,68 @@ async def _execBackupTask(self, dirn):
...
-            logger.debug('Acquiring LMDB txns')
-            mypipe.send('proceed')
+                logger.debug('Starting backup process')
 
-            # This is technically pending the ioloop waiting for the backup process to acquire a bunch of
-            # transactions.  We're effectively locking out new write requests the brute force way.
-            hasdata = mypipe.poll(timeout=self.BACKUP_ACQUIRE_TIMEOUT)
-            if not hasdata:
-                raise s_exc.SynErr(mesg='backup subprocess stuck acquiring LMDB txns')
+                args = (child_pipe, self.dirn, dirn, paths, loglevel)
```
Comment (invisig0th) [inline synapse/lib/cell.py:1219]: would it make sense to add this to the executor?

**Review by invisig0th — CHANGES_REQUESTED** [review]

**Review by invisig0th — APPROVED** [review]

File: synapse/lib/cell.py [hunk 2]
```diff
@@ -1202,75 +1201,69 @@ async def _execBackupTask(self, dirn):
...
 
-            data = mypipe.recv()
-            assert data == 'captured'
+                def waitforproc1():
+                    nonlocal proc
```
Comment (vEpiphyte) [inline synapse/lib/cell.py:1222]: Can we return the proc object ?

**Comment by vEpiphyte** [comment]
Broken due to https://github.com/jupyter/jupyter_client/issues/637

File: synapse/lib/cell.py (hunk 2 above)
Comment (invisig0th) [reply synapse/lib/cell.py:1222]: This seems like a good simplification. :)

Comment (jnwatson) [reply synapse/lib/cell.py:1222]: That won't kill the process if it gets stuck in waitforproc1.

**Pull request merged**